frontend/src/metabase/static-viz/components/LineAreaBarChart/utils/series.unit.spec.ts

```diff
@@ -0,0 +1,378 @@
+  },
+};
+
+describe("getSeriesWithColors", () => {
+  it("should return an empty series given an empty series", () => {
+    const seriesWithColors = getSeriesWithColors([], getPalette({}), settings);
+
+    expect(seriesWithColors).toEqual([]);
+  });
+
+  describe("Series without breakout values", () => {
+    const series: SeriesWithoutBreakoutValues[] = [
+      {
+        name: "Count",
+        yAxisPosition: "left",
+        type: "bar",
+        data: [
+          ["Doohickey", 3976],
+          ["Gadget", 4939],
+          ["Gizmo", 4784],
+          ["Widget", 5061],
+        ],
+        seriesKey: "count",
+      },
+    ];
+
+    it("should assign colors given series", () => {
+      const seriesWithColors = getSeriesWithColors(
+        series,
+        getPalette({}),
+        settings,
+      );
+
+      const expectedSeries = [
+        {
+          name: "Count",
+          color: "#509EE3", // brand color
+          yAxisPosition: "left",
+          type: "bar",
+          data: [
+            ["Doohickey", 3976],
+            ["Gadget", 4939],
+            ["Gizmo", 4784],
+            ["Widget", 5061],
+          ],
+        },
+      ];
+
+      expect(seriesWithColors).toEqual(expectedSeries);
+    });
+
+    it("should assign colors from whitelabel colors", () => {
+      const seriesWithColors = getSeriesWithColors(
+        series,
+        getPalette({ brand: "#123456", summarize: "#ffffff" }),
+        settings,
+      );
+
+      const expectedSeries = [
+        {
+          name: "Count",
+          color: "#123456", // whitelabel color
+          yAxisPosition: "left",
+          type: "bar",
+          data: [
+            ["Doohickey", 3976],
+            ["Gadget", 4939],
+            ["Gizmo", 4784],
+            ["Widget", 5061],
+          ],
+        },
+      ];
+
+      expect(seriesWithColors).toEqual(expectedSeries);
+    });
+
+    it("it should assign colors from column colors", () => {
+      const seriesWithColors = getSeriesWithColors(
+        series,
+        getPalette({ brand: "#123456", summarize: "#ffffff" }),
+        merge(settings, {
+          series_settings: {
+            count: {
+              color: "#987654",
+            },
+          },
+        }),
+      );
+
+      const expectedSeries = [
+        {
+          name: "Count",
+          color: "#987654", // column color
+          yAxisPosition: "left",
+          type: "bar",
+          data: [
+            ["Doohickey", 3976],
+            ["Gadget", 4939],
+            ["Gizmo", 4784],
+            ["Widget", 5061],
+          ],
+        },
+      ];
+
+      expect(seriesWithColors).toEqual(expectedSeries);
+    });
+  });
+
+  describe("Series with preferred colors", () => {
+    const series: SeriesWithoutBreakoutValues[] = [
+      {
+        name: "Sum of Total",
+        yAxisPosition: "left",
+        type: "bar",
+        data: [["2016-04-24T00:00:00Z", 52.75594257942132]],
+        seriesKey: "sum",
```

Comment:
for a series key with value that's defined as one of the option in this function https://github.com/metabase/metabase/blob/a4255c5e749bfc3053f20d3c64010055ae349917/frontend/src/metabase/lib/colors/groups.ts#L41

it would get a predetermined color